gems/sorbet-runtime/lib/types/struct.rb

```diff
@@ -2,9 +2,19 @@
 # typed: true
 
 class T::InexactStruct
+  extend T::Sig
+
   include T::Props
   include T::Props::Serializable
   include T::Props::Constructor
+
+  sig {params(other: Object).returns(T::Boolean)}
+  def ==(other)
```

Comment:
This would be super slow performance-wise and have surprising behaviors(e.g. it can easily go into infinite loop).

WDYT about making this implementation into a mixin  while it gets stabilized?
This is a very convenient operation and people would be tempted to use it, and we should be careful before introducing it. E.g., similarly how getters & setters specialize themselves to a particular struct-class, we should probably specialize comparison funcition

Reply:
E.g. This `==` doesn't agree with hash. There are quite a few errors to iron out

Reply:
Making this a mixin sounds fairly reasonable to me, that's a good suggestion

As for the infinite looping, it seems like that's already an issue with Struct, no? I just tested out with:
```
class Box < T::Struct
  prop :value, T.nilable(Object)
end
     
box1 = Box.new
box2 = Box.new
box1.value = box2
box2.value = box1
```

And the `to_str` and `serialize` methods both result in `ThreadErrors`

Reply:
Not sure I understand the `This == doesn't agree with hash` comment, could you explain that further?